login/src/main/java/com/fightpandemics/login/ui/LoginViewModel.kt

```diff
@@ -136,12 +178,117 @@ class LoginViewModel @Inject constructor(
                 }
             }
         }
+    }
 
+    // Get user location from API using lat & lng
+    fun updateCurrentLocation(location: Location) {
+        sharelocationJob?.cancel()
+        sharelocationJob = viewModelScope.launch {
+            userLocationUseCase(LocationRequest(location.latitude, location.longitude))
+                .collect {
+                    when (it) {
+                        is Result.Loading -> currentLocation(true, null, null)
+                        is Result.Success -> currentLocation(
+                            false,
+                            null,
+                            it.data as com.fightpandemics.core.data.model.userlocation.Location
+                        )
+                        is Result.Error -> currentLocation(true, it, null)
+                    }
+                }
+        }
+    }
+
+//     Search for location from API using user input
+    @FlowPreview
+    fun searchLocation() {
+        searchlocationJob?.cancel()
+        searchlocationJob = viewModelScope.launch {
+            searchQuery
+                .debounce(300)
+                .filter { return@filter it.isNotEmpty() && it.length >= LEN_FOR_SUGGESTIONS }
+                .distinctUntilChanged()
+                .map { it.trim() }
+                .flatMapLatest { locationPredictionsNameAndIdUseCase(it) }
+                .conflate()
+                .collect {
+                    when (it) {
+                        is Result.Success -> {
+                            _searchLocationState.value =
+                                it.data as MutableList<Prediction>
+                        }
+                        is Result.Error -> Timber.e(it.toString())
+                        is Result.Loading -> Timber.e("LOADING...")
+                    }
+                }
+        }
+    }
+
+    private fun currentLocation(
+        loading: Boolean,
+        error: Result.Error?,
+        currentLocation: com.fightpandemics.core.data.model.userlocation.Location?
+    ) {
+
+        if (currentLocation != null) {
+            completeProfileLocation = com.fightpandemics.core.data.model.login.Location(
+                currentLocation.address ?: "",
```

Comment:
See comment above about nullable fields in `Location`. If they were not nullable these checks wouldn't be needed right?

Reply:
Location data from the server does not guarantee that it will return the field. The rest of the explanation was above.